assets/src/edit-story/app/prepublish/errors/metadata.js

```diff
@@ -0,0 +1,167 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import isElementBelowLimit from '../../../utils/isElementBelowLimit';
+import { PRE_PUBLISH_MESSAGE_TYPES } from '../constants';
+
+const FEATURED_MEDIA_RESOURCE_MIN_HEIGHT = 853;
+const FEATURED_MEDIA_RESOURCE_MIN_WIDTH = 640;
+
+const PUBLISHER_LOGO_MIN_HEIGHT = 96;
+const PUBLISHER_LOGO_MIN_WIDTH = 96;
+
+/**
+ *
+ * @typedef {import('../types').Guidance} Guidance
+ * @typedef {import('../../../types').Story} Story
+ */
+
+/**
+ * Check the story for a cover.
+ * If the story does not have a cover, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function storyCoverAttached(story) {
+  if (typeof story.featuredMediaUrl !== 'string') {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __('Missing story cover', 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check the story for a title.
+ * If the story does not have a title, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function storyTitle(story) {
+  if (typeof story.title !== 'string' || story.title.trim() === '') {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __('Missing story title', 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check that the story's cover resource has sufficient dimensions.
+ * If the resource is too small in either dimension, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function storyCoverPortraitSize(story) {
+  if (
+    story.featuredMedia.height < FEATURED_MEDIA_RESOURCE_MIN_HEIGHT ||
+    story.featuredMedia.width < FEATURED_MEDIA_RESOURCE_MIN_WIDTH
+  ) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __("Story's portrait cover image is too small", 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check that the story's publisher logo resource has sufficient dimensions.
+ * If the resource is too small in either dimension, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function publisherLogoSize(story) {
+  if (
+    story.publisherLogo.height < PUBLISHER_LOGO_MIN_HEIGHT ||
```

Comment:
Is `story.publisherLogo` the right path for this? I see `featuredMedia` gets sent down, but I don't see any reference to `publisherLogo` getting passed through for testing.

Reply:
yeah check out the update story calls [here for featuredMedia](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/inspector/document/publish/index.js#L95) and [here for publisherLogo](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/inspector/document/publish/index.js#L107) (`updateStory` reducer is implemented [here](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/app/story/useStoryReducer/reducers/updateStory.js). They both live on the top-level story and will probably be migrated to objects in a similar way, e.g. https://github.com/google/web-stories-wp/issues/5105

We may end up removing this check entirely in favor of [just preventing the upload of a logo that is too small](https://github.com/google/web-stories-wp/issues/5107)